### ISSUE TITLE
Metadata: Option to remove existing  location estimates

### ISSUE THREAD

**Issue by HeikoGr** [photoprism/photoprism#3232]
there should be an option to disable the placement of pictures with estimated locations on the map.

i was able to remove the pictures from the map by updateing the database as described here: #1395 (i did set lat and lon to 0, too).

**Describe what problem this solves and why this would be valuable to many users**
my expectation was that the pictures would disappear from the map after i unchecked the option to estimate locations.
Further thinking brought to my mind, that the actual option is related to indexing - and is no map setting.

**Comment by graciousgrey** [comment]
The map shows all photos with lat/lng information. If you disable the location estimates this does not remove already existing estimates.

**Comment by HeikoGr** [comment]
Yes, and it works as designed. But it would be nice to have the ability to either

- remove the estimates from the database, or
- disable the mapping of the photos with estimated locations

**Comment by graciousgrey** [comment]
I guess removing estimates makes more sense in this case or what do you think?

**Comment by HeikoGr** [comment]
In my opinion: yes, removing the estimates does make more sense.

**Comment by lastzero** [comment]
AFAIK the background worker should remove the estimates from the pictures. However, this does not happen instantly, but over time.

**Comment by HeikoGr** [comment]
I will test it and report my mileage.

reading through the source (i might be totally wrong - i am bad at programming) in /internal/entity/photo_optimize.go it‘s only updating if ~lat/lon is set or~ the estimate setting is checked.

**Comment by HeikoGr** [comment]
I tested it the following way:
- make shure there is no estimated geolocation in the database (`select * from photos where place_src = 'estimate'` => 0 entries)
- enable option to estimate location
- run `docker compose exec photoprism photoprism optimise -f`
- check database: now there are 2.430 entries with estimated locations
- disable option to estimate location
- restart photoprism and make shure the option is still disabled (just to be shure)
- run `docker compose exec photoprism photoprism optimise -f`
- check database: there are still 2.430 entries with estimated locations

so i might be right with my assumption, that the optimizer is only able to set/update estimated locations. 
It's actually not able to unset.

as this should be a really small change i would love to make a PR in the next days/weeks by myself. (including documentation and test case)

**Comment by lastzero** [comment]
It can take up to 7 days. I can also verify and fix this if needed, however we are currently out of the office.

**Comment by HeikoGr** [comment]
I thought the optimize command does the same as the background worker?

It's not super important for me (i found my fix by alter the database) - i just thought the actual behaviour may not meet the normal user expectation.

**Comment by lastzero** [comment]
Maybe if you run it with --force. Don't remember all the options.

**Comment by lastzero** [comment]
Seems I was right about the 7 days (check was broken, but is fixed with the most recent patch):

<img width="883" alt="Screenshot 2023-03-03 at 20 41 00" src="https://user-images.githubusercontent.com/301686/222811778-5e7404e6-66f4-4a3d-b76b-27ff80079130.png">

**Comment by HeikoGr** [comment]
@lastzero thank you for your last fix, as for now, you don't have to take action - i am still at research

---

little update from me on my research, even if the 7 days ar not over yet:

> Maybe if you run it with --force. Don't remember all the options.

to be fair - the command is not documented. I got it from [https://www.reddit.com/r/photoprism/comments/sec7y9/location_info/](url)
as far as i understand the optimize command is only part of the background worker.

the worker does more then metadata:
 - metadata worker (<- runs optimize)
 - share worker
 - sync worker

but i can't find any code where the estimate location got deleted if the option is disabled.

back to my test database:
- all 2430 estimated fotos are still in the database
- the database entries do all have recent updates (fields "checked_at" and "estimated_at")

so i still doubt that the estimates gets deleted at the moment, but i will continue research.

**Comment by lastzero** [comment]
Still traveling and on my phone. Once I'm back at my workstation I can sort it out for very quickly.

**Comment by HeikoGr** [comment]
I can now for sure say that the background worker did not remove the estimated locations after unsetting the index feature "estimates".

I installed a clean instance of photoprism 3 weeks ago and set "estimates" before importing the pictures (and disabled it afterwards).

@lastzero  would you like to prefer to look after it for yourself or should i try to write a PR? I was not able to set up a development environment at the moment and i am not sure which way you would prefer:

- delete the estimates from the db at the moment the setting was updated
- let a worker do the job
- keep the metadata in the database and disable the mapping of estimated places

**Comment by dd900** [comment]
I have this same problem. There is no way I'm going to delete and reinstall. I've had to tag too many faces manually. I installed using Portainer and I can not access the MariaDB database. Manually entering 0 into Lat and Lon does nothing in the UI. It still retains the location. This feature should NOT be enabled by default.

this command does not work `docker-compose exec mariadb mysql -u root -p [root password] photoprism` from the compose.yml dir. Says MariaDB is down (it's not)

running `docker exec mariadb mysql -u root -p [root password] photoprism` from terminal  just prints the database help dialog

**Comment by lastzero** [comment]
@dd900 If you need help, you are welcome to post your question in [GitHub Discussions](https://link.photoprism.app/discussions) or ask in our [Community Chat](https://link.photoprism.app/chat). Common problems can be quickly diagnosed and solved using our [Troubleshooting Checklists](https://docs.photoprism.app/getting-started/troubleshooting/). [Silver, Gold, and Platinum](https://link.photoprism.app/membership) members are also welcome to email us for technical support and advice.

[View Support Options ›](https://www.photoprism.app/kb/getting-support)

**Comment by HeikoGr** [comment]
your exec command is wrong. I need to run
`docker exec -it photoprism-mariadb-1 bash ` if i want to run the bash shell interactively on the mariadb container. From there you can run mysql/mariadb shell commands.

**Comment by dd900** [comment]
> your exec command is wrong. I need to run `docker exec -it photoprism-mariadb-1 bash ` if i want to run the bash shell interactively on the mariadb container. From there you can run mysql/mariadb shell commands.

Thank you I'm in but now how do I execute this statement? `UPDATE photos SET place_id = 'zz', place_src = '' WHERE place_src = 'estimate';`

**Comment by lastzero** [comment]
We've recently added docs that explain how to open a terminal with Portainer: https://docs.photoprism.app/getting-started/portainer/#command-line-interface

Anyhow, it would be best to discuss this on GitHub Discussions and not in the issue comments where everyone gets an email notification for every new comment. Thank you! :)

**Comment by dd900** [comment]
> We've recently added docs that explain how to open a terminal with Portainer: https://docs.photoprism.app/getting-started/portainer/#command-line-interface

??? That's normal for every Portainer container and the docs do not explain how to interact with the database.

> Anyhow, it would be best to discuss this on GitHub Discussions and not in the issue comments where everyone gets an email notification for every new comment. Thank you! :)

Anyhow since my issue is still not resolved...

**Comment by lastzero** [comment]
How complicated is it to follow the link and simply ask there????

**Comment by dd900** [comment]
> How complicated is it to follow the link and simply ask there????

The link is to a documentation page. You can't ask questions there... I made a discussion. My bad. I'll never report that I'm having the same issue as other users again. Your condescending and unhelpful replies are why I will always be a free user.

**Comment by lastzero** [comment]
👉  https://github.com/photoprism/photoprism/discussions
